services/src/main/java/org/keycloak/protocol/saml/mappers/SAMLLoginResponseMapper.java

```diff
@@ -29,6 +29,7 @@
  */
 public interface SAMLLoginResponseMapper {
 
+
     ResponseType transformLoginResponse(ResponseType response, ProtocolMapperModel mappingModel, KeycloakSession session,
-                                        UserSessionModel userSession, AuthenticatedClientSessionModel clientSession);
+                                        UserSessionModel userSession, ClientSessionContext clientSessionCtx);
```

Comment:
Why is this change here?

Reply:
This change is because the audience resolve mapper needs to compute the groups of the user in each client (the resolve mapper duplicates the OIDC behavior, and the clientId is added in the audience if the user has at least one group in that client). So for that I re-used the [clientSessionCtx.getRoles()](https://github.com/keycloak/keycloak/pull/6520/files#diff-ec80ecf52bea4386f9a2e6e04c51d6b1R85) method. This class is also passed to the [SAMLRoleListMapper](https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/protocol/saml/mappers/SAMLRoleListMapper.java#L32). If I hadn't passed it I would have needed to calculate groups again (repeating code and calculation), as I saw no mapper used this interface I decided to modify it (the previous AuthenticatedClientSessionModel can also be accessed through the ClientSessionContext).

Reply:
Thanks for explanation, it really makes sense to keep this on par with `OIDCAccessTokenMapper`.